CONTRIBUTING.md

```diff
@@ -1,9 +1,190 @@
+
+In order to contribute, you will need to do the following:
+
+1) Create your own branch
+2) Make sure that tests pass and code coverage is maintained
+3) Open a Pull Request
+
+The NeMoS package follows the [GitHub Flow](https://www.gitkraken.com/learn/git/best-practices/git-branch-strategy#github-flow-branch-strategy) workflow. In essence, this means no one is allowed to
+push to the `main` branch and all development happens in separate feature branches that are then merged into `main` once we have determined they are ready. When enough changes have accumulated, we 
+put out a new release, adding a new tag which increments the version number, and upload the new release to PyPI. 
+
+
+#### Creating a development environment
+
+You will need a local installation of `nemos` which keeps up-to-date with any changes you make. To do so, you will need to clone `nemos` and checkout a new branch:
+
+1) Clone the `nemos` repo to your local machine
+
+```bash
+git clone https://github.com/flatironinstitute/nemos.git
+cd nemos
+```
+
+2) Install `nemos` in editable mode with developer dependencies  
+```bash
+pip install -e .[dev]
+```
+
+> **_NOTE:_**  In order to install `nemos` in editable mode you will need a Python virtual environment. Please see our documentation [here](https://nemos.readthedocs.io/en/latest/installation/) that 
+> provides guidance on how to create and activate a virtual environment.
+
+
+#### Creating a new branch
+
+As mentioned previously, each feature in `nemos` is worked on in a separate branch. This allows multiple people developing multiple features simultaneously, without interfering with each other's work. To create
+your own branch, run the following from within your `nemos` directory:
+
+```bash
+# switch to the main branch
+git checkout main
+# update your main branch
+git pull origin main
+# create and switch to a new branch
+git checkout -b my_feature_branch
+```
+
+After you have made changes on this branch, add and commit them when you are ready:
+
+```bash
+# stage the changes
+git add src/nemos/the_changed_file.py
+# commit your changes
+git commit -m "A one-line message explaining the changes made"
+# push to the remote origin
+git push origin my_feature_branch
+```
+
+#### Contributing your change back to NeMoS
+
+You can make any number of changes on your branch. Once you are happy with your changes, add tests to check that they run correctly and add documentation to properly note your changes. 
+See below for details on how to [add tests](#adding-tests) and properly [document](#adding-documentation) your code.
+
+Lastly, you should make sure that the existing tests all run successfully and that the codebase is formatted properly:
+
+```bash
+# run tests and make sure they all pass
+pytest tests/
+# format the code base
+black src/
+black tests/
+```
+> **_NOTE:_** If some files were reformatted after running `black`, make sure to commit those changes and push them to your feature branch as well. 
```

Comment:
```suggestion

> **_INFO:_** [`black`](https://black.readthedocs.io/en/stable/) and [`isort`](https://pycqa.github.io/isort/) automatically reformat your code and organize your imports, respectively. [`flake8`](https://flake8.pycqa.org/en/stable/#) does not modify your code directly; instead, it identifies syntax errors and code complexity issues that need to be addressed manually.

> **_NOTE:_** If some files were reformatted after running `black` or `isort`, make sure to commit those changes and push them to your feature branch as well. 
```

I don't know if the `__INFO:__` renders well, but we need to provide some details about the fact that if `flake8` fails, errors must be addressed.